server/api/src/main/java/org/hawkular/apm/server/api/utils/SpanUniqueIdGenerator.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.server.api.utils;
+
+import org.hawkular.apm.server.api.model.zipkin.Span;
+
+/**
+ * Spans generally have an unique id, except when there are two spans representing the client and server
+ * side of a communication, in which case they share the same span id. Therefore, it is necessary to
+ * provide an utility class for changing the id to avoid overriding entities in a database, cache...
+ * To the id of the client spans is added a string suffix.
+ *
```

Comment:
Thanks for the JavaDoc! Is it possible that a Span will be used by N clients and 1 server? In other words: is it possible to have two "abc-client" IDs? If so, what are the consequences?